src/collections/Menu/Menu.js

```diff
@@ -145,28 +145,37 @@ class Menu extends Component {
   static Item = MenuItem
   static Menu = MenuMenu
 
-  renderItems() {
-    const { items, onItemClick } = this.props
+  componentWillMount() {
+    super.componentWillMount()
 
-    this.trySetState({ activeIndex: _.findIndex(items, ['active', false]) || -1 })
+    const { items } = this.props
+    if (items) this.trySetState({ activeIndex: _.findIndex(items, ['active', false]) || -1 })
+  }
 
-    return _.map(items, (index, item) => {
-      const { content, childKey, name, onClick, itemProps } = item
+  handleItemClick(e, { name, index }) {
+    const { onItemClick } = this.props
 
-      const active = this.state.activeIndex === index
-      const finalKey = childKey || [content, name].join('-')
+    this.trySetState({ activeIndex: index })
+    if (onItemClick) onItemClick(e, { name, index })
+  }
 
-      const handleClick = (e, { itemName, itemIndex }) => {
-        const handler = onItemClick || onClick
-        this.trySetState({ activeIndex: itemIndex })
+  renderItems() {
+    const { items } = this.props
+    const { activeIndex } = this.state
 
-        if (handler) handler(e, { itemName, itemIndex })
+    return _.map(items, (item, index) => {
+      const { content, childKey, name, onClick, itemProps } = item
+      const finalKey = childKey || [content, name].join('-')
+
+      const handleClick = (e, data) => {
+        this.handleItemClick(e, data)
+        if (onClick) onClick(e, data)
       }
```

Comment:
I'm moved condition to `handleItemClick`.

```js
handleItemClick(e, { name, index }, onClick) {
  ...
  if (onClick) onClick(e, { name, index })
  if (onItemClick) onItemClick(e, { name, index })
}
```

But problem will be still there, `handleClick` will be created on each iteration, or not?

```jsx
const handleClick = (e, data) => this.handleItemClick(e, data, onClick)
return <MenuItem onClick={handleClick} />
//
return <MenuItem onClick={(e, data) => this.handleItemClick(e, data, onClick)} />
```